internal/controller/eventing/controller.go

```diff
@@ -116,12 +132,112 @@ func (r *Reconciler) handleEventingDeletion(_ context.Context, _ *eventingv1alph
 }
 
 func (r *Reconciler) handleEventingReconcile(ctx context.Context,
-	_ *eventingv1alpha1.Eventing, log *zap.SugaredLogger) (ctrl.Result, error) {
+	eventing *eventingv1alpha1.Eventing, log *zap.SugaredLogger) (ctrl.Result, error) {
 	log.Info("handling Eventing reconciliation...")
-	// TODO: Implement me.
 
-	// just to use the variables.
-	log.Info(FinalizerName, ManagedByLabelKey, ManagedByLabelValue)
+	// set state processing if not set yet
+	r.InitStateProcessing(eventing)
+
+	for _, backend := range eventing.Spec.Backends {
+		switch backend.Type {
+		case eventingv1alpha1.NatsBackendType:
+			return r.reconcileNATSBackend(ctx, eventing, log)
+		case eventingv1alpha1.EventMeshBackendType:
+			return r.reconcileEventMeshBackend(ctx, eventing, log)
+		default:
+			return ctrl.Result{Requeue: false}, fmt.Errorf("not supported backend type %s", backend.Type)
+		}
+	}
+	// this should never happen, but if happens do nothing
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) reconcileNATSBackend(ctx context.Context, eventing *eventingv1alpha1.Eventing, log *zap.SugaredLogger) (ctrl.Result, error) {
+	// check nats CR if it exists and is in natsAvailable state
+	err := r.checkNATSAvailability(ctx, eventing)
+	if err != nil {
+		return ctrl.Result{}, r.syncStatusWithNATSErr(ctx, eventing, err, log)
+	}
 
+	// set NATSAvailable condition to true and update status
+	eventing.Status.SetNATSAvailableConditionToTrue()
+	if err := r.syncEventingStatus(ctx, eventing, log); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	deployment, err := r.handlePublisherProxy(ctx, eventing, log)
+	if err != nil {
+		return ctrl.Result{}, r.syncStatusWithPublisherProxyErr(ctx, eventing, err, log)
+	}
+
+	return r.handleEventingState(ctx, deployment, eventing, log)
+}
+
+func (r *Reconciler) checkNATSAvailability(ctx context.Context, eventing *eventingv1alpha1.Eventing) error {
+	natsAvailable, err := r.eventingManager.IsNATSAvailable(ctx, eventing.Namespace)
```

Comment:
Is it confirmed that both CRs (Eventing & NATS) will be deployed inside the same namespace?

Reply:
good question. As for [proposal](https://github.tools.sap/skydivingtunas/proposals/blob/main/eventing-modularization/eventing-crd.md), we don't specify where NATS resources are located. Moreover, we've decided to restrict NATS CR namespace and name to `kyma-system` and `eventing-nats` respectively. This means implicitly Eventing CR should be deployed there. 

Is it okay if we keep the current change and enhance it afterwards? I'll bring the point to grooming/tech talk next week.